facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigurationProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.facade.providers;
+
+import org.jboss.pnc.dto.BuildConfiguration;
+import org.jboss.pnc.dto.BuildConfigurationRef;
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.facade.mapper.api.BuildConfigurationMapper;
+import org.jboss.pnc.facade.rsql.mapper.BuildConfigurationRSQLMapper;
+import org.jboss.pnc.spi.datastore.repositories.BuildConfigurationRepository;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+
+import static org.jboss.pnc.spi.datastore.predicates.BuildConfigurationPredicates.withProductVersionId;
+
+@Stateless
+public class BuildConfigurationProvider
+        extends AbstractProvider<org.jboss.pnc.model.BuildConfiguration, BuildConfiguration, BuildConfigurationRef> {
+
+    // needed for CDI / EJB
+    public BuildConfigurationProvider() {}
```

Comment:
In the provider, this should not be needed. (Less so with public constructor!) Instead create interface for the BuildConfigurationProvider and inject that interface.

(CDI must be able to create proxy for the thing (type) you have with your `@Inject` annotation. The proxy is inherited from the injected type, so the type must be non-final and **either interface** or a class with no-args constructor.)

Reply:
The same goes for the other providers and the abstract provider.

Reply:
Meh seems time consuming to create an interface just for this to work without a default constructor.

Reply:
right click -> refactor -> extract -> interface :P

Reply:
k fine!